src/stdlib/format_timestamp.rs

```diff
@@ -43,39 +53,66 @@ impl Function for FormatTimestamp {
     ) -> Compiled {
         let value = arguments.required("value");
         let format = arguments.required("format");
+        let tz: Option<Box<dyn Expression>> = arguments.optional("tz");
 
-        Ok(FormatTimestampFn { value, format }.as_expr())
+        Ok(FormatTimestampFn { value, format, tz }.as_expr())
     }
 
     fn examples(&self) -> &'static [Example] {
-        &[Example {
-            title: "format timestamp",
-            source: r#"format_timestamp!(t'2021-02-10T23:32:00+00:00', "%d %B %Y %H:%M")"#,
-            result: Ok("10 February 2021 23:32"),
-        }]
+        &[
+            Example {
+                title: "format timestamp",
+                source: r#"format_timestamp!(t'2021-02-10T23:32:00+00:00', format: "%d %B %Y %H:%M")"#,
+                result: Ok("10 February 2021 23:32"),
+            },
+            Example {
+                title: "format timestamp with tz",
+                source: r#"format_timestamp!(t'2021-02-10T23:32:00+00:00', format: "%d %B %Y %H:%M", tz: "Europe/Berlin")"#,
+                result: Ok("11 February 2021 00:32"),
+            },
+        ]
     }
 }
 
 #[derive(Debug, Clone)]
 struct FormatTimestampFn {
     value: Box<dyn Expression>,
     format: Box<dyn Expression>,
+    tz: Option<Box<dyn Expression>>,
 }
 
 impl FunctionExpression for FormatTimestampFn {
     fn resolve(&self, ctx: &mut Context) -> Resolved {
         let bytes = self.format.resolve(ctx)?;
         let ts = self.value.resolve(ctx)?;
 
-        format_timestamp(bytes, ts)
+        match self.tz.clone() {
+            None => format_timestamp_with_tz(bytes, ts, &Utc),
+            Some(tz) => {
+                let tz = &tz.resolve(ctx)?.try_bytes()?;
+                let tz = String::from_utf8_lossy(tz);
+                match tz {
+                    std::borrow::Cow::Borrowed("Local") => {
+                        format_timestamp_with_tz(bytes, ts, &Local)
```

Comment:
For `Local` I think we'll actually want to use the `timezone` option on the `remap` transform if it has been specified. You can get that from the `ctx` object like you see in `parse_timestamp`:

https://github.com/vectordotdev/vrl/blob/43df3373c03125d53ee1638909227ce9eca66955/src/stdlib/parse_timestamp.rs#L8

Reply:
Took the `*ctx.timezone()`, but slightly different way than was suggested.